src/state/swap.ts

```diff
@@ -27,3 +27,6 @@ export const swapAtom = atomWithImmer<Swap>({
 export const displayTxHashAtom = atom<string | undefined>(undefined)
 
 export const feeOptionsAtom = atom<FeeOptions | undefined>(undefined)
+
+// If set, allows integrator to add behavior when 'Connect wallet to swap' button is clicked
+export const onConnectWalletClickAtom = atom<(() => void | Promise<boolean>) | undefined>(undefined)
```

Comment:
not sure if this is the right file to put this atom in - but not sure where else to put it

Reply:
i would make a `wallet.ts` for wallet state